### PR TITLE
Simplify/document snap creation

### PR DESCRIPTION
**Pull request by ohemorange**
This PR adds the following documentation improvements to fix https://github.com/certbot/certbot/issues/7958:

- Simplify building external plugins
- Separate out certbot snap instructions from plugin instructions
- Mention that dnsimple is just an example for the plugin instructions
- Mention remote build for other architectures
- Mention snap doc exists elsewhere in developer guide (`contributing.rst`)

We could more neatly create the constraints file, as described in https://github.com/certbot/certbot/issues/8100, but what I did here achieves the goal of making it easier for someone building the snaps manually, since we probably do want to re-generate everything anyway. I think improving it for its own sake can come later.

Also, remote build doesn't seem to be launching the build when I tested the instructions, not sure what's going on there. I was going to wait and see if it works when azure tries.

Modified docs sections look as follows:
Screenshot from 2020-10-23 16-29-25
Screenshot from 2020-10-23 16-29-08

### CODE REVIEW

**Comment by bmw** [comment]
Closing and reopening to fix the CI status checks.

**Review by bmw — CHANGES_REQUESTED** [review]
Thanks a lot for working on this!

> Also, remote build doesn't seem to be launching the build when I tested the instructions, not sure what's going on there.

I think what's going on here is this command is really slow if you don't have a clean clone of our repo. `snapcraft` sends everything under your CWD to Launchpad and we often have things like multiple virtual environments (each of which is ~200MB), multiple clones of the repo under `releases/` or `tests/letstest/`, etc.

You can see this by running `snapcraft remote-build` directly with a command like `snapcraft remote-build --build-on=amd64` from the root of our repo to build the Certbot snap. On a clean clone on my laptop, in about a minute it prints:
```
Packaging sources for certbot...
Packaging sources for shared-metadata...
Sending build data to Launchpad... (https://bradmwarren:<token>@git.launchpad.net/~bradmwarren/+git/snapcraft-certbot-06e0c4/)
Enumerating objects: 12, done.
Counting objects: 100% (12/12), done.
Delta compression using up to 4 threads
Compressing objects: 100% (10/10), done.
Writing objects: 100% (12/12), 59.22 MiB | 938.00 KiB/s, done.
Total 12 (delta 1), reused 0 (delta 0), pack-reused 0
To https://git.launchpad.net/~bradmwarren/+git/snapcraft-certbot-06e0c4/
 * [new branch]      HEAD -> master
If interrupted, resume with: 'snapcraft remote-build --recover'
Building snap package for amd64. This may take some time to finish.
```
In the clone of the Certbot repo I normally use it hangs on the earlier steps much longer.

Assuming this was the problem for you, I think adding a sentence about this to our docs is a good idea.

**Comment by ohemorange** [comment]
Ah yes, that's almost certainly what it was, running the raw command now. I did not know this, I'll add a sentence about that!

File: certbot/docs/contributing.rst [hunk 1]
```diff
@@ -374,6 +374,8 @@ identifier ``metadata-1``.
 The script used to generate the snapcraft.yaml files for our own externally
 snapped plugins can be found at
 https://github.com/certbot/certbot/blob/master/tools/snap/generate_dnsplugins_snapcraft.sh.
+Instructions for how to build our externally snapped plugins can be found in the README file at
+https://github.com/certbot/certbot/tree/master/tools/snap.
```
[the remark this answers is not in the document]
Comment (ohemorange) [reply certbot/docs/contributing.rst:378]: great idea!

File: tools/snap/generate_dnsplugins_postrefreshhook.sh [hunk 2]
```diff
@@ -1,13 +1,13 @@
 #!/bin/bash
-# Generate the hooks/post-refresh file for all DNS plugins
+# Generate the hooks/post-refresh file for a DNS plugin
+# Usage: bash generate_dnsplugins_postrefreshhook.sh path/to/dns/plugin
+# For example, from the certbot home directory:
+#   bash generate_dnsplugins_postrefreshhook.sh certbot-dns-dnsimple
```
[the remark this answers is not in the document]
Comment (ohemorange) [reply tools/snap/generate_dnsplugins_postrefreshhook.sh:5]: I'll do it as `tools/snap/` for ease of copy-pasting!

File: tools/snap/README.md [hunk 3]
```diff
@@ -15,31 +18,78 @@ These steps need to be done once to set up your VM and do not need to be run aga
...
+
+### Initial Local Setup
+
+ 1. Create or log into an Ubuntu One account [here](https://login.launchpad.net/).
+ 2. Install git and python with `sudo apt update && sudo apt install -y git python`.
+ 3. Install snapcraft with `sudo snap install --classic snapcraft`.
+ 4. `cd ~` (or any other directory where you want our source files to be)
+ 5. Run `git clone git://github.com/certbot/certbot`
+ 6. `cd certbot` (All further instructions are relative to this directory.)
+
```
[the remark this answers is not in the document]
Comment (ohemorange) [reply tools/snap/README.md:85]: Oh, I'd actually assumed that you had to authorize it for each one you're building, this suggestion is much better!

**Comment by ohemorange** [comment]
Here's the new `contributing.rst` section; the link works.

Screenshot from 2020-10-26 16-46-16

**Review by bmw — APPROVED** [review]
Looks great! Thanks again for grabbing this.

**Pull request merged**